lib/exceptional_synchrony/callback_exceptions.rb

```diff
@@ -38,6 +38,40 @@ def return_exception(*args)
           ex
         end
       end
+
+      private
+
+      def handle_result_error(result)
+        error = result.error
+        if error_is_a_timeout?(error)
+          raise Timeout::Error
+        elsif error_is_a_syscall_class?(error)
+          raise Failure, format_failure(error.to_s, format_inspect(result))
+        else
+          raise Failure, format_failure(format_inspect(error), format_inspect(result))
+        end
+      end
+
+      def format_failure(error_string, result_string)
+        "ERROR = #{error_string}; RESULT = #{result_string}"
+      end
+
+      def format_inspect(obj)
```

Comment:
Maybe name this `truncated_inspect`? And as I mentioned in person, I like the `,` version of the slice notation because the exact length is visible without the off-by-one bit.
```
inspection = obj.inspect[0, 101]
...
```